server/testserver.go

```diff
@@ -45,8 +47,9 @@ func StartTestServer(t *testing.T) *TestServer {
 
 // NewTestContext returns a context for testing. It overrides the
 // Certs with the test certs directory.
-// TODO(marc): use embedded certs.
+// We need to override the certs loader.
 func NewTestContext() *Context {
+	security.SetReadFileFn(securitytest.Asset)
```

Comment:
wherever this mock call is coming from, it definitely needs to be in the tests. So you'll need to put the `main_test.go` files back, I think.